### PR TITLE
#9Task. Mikhail Pyshnik

### PR DESCRIPTION
**Pull request by MikhailPyshnik**
https://travis-ci.org/github/MikhailPyshnik/js-assignments/builds/704886572

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -26,7 +26,10 @@
  *
  */
 function getComposition(f,g) {
-    throw new Error('Not implemented');
+    return function(x) 
+    {
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: Curly bracket should be opened on the same line as function or expression declaration

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -65,10 +71,22 @@ function getPowerFunction(exponent) {
...
+        {
+            let totalSum = 0;
+            for (let i = 0; i < args.length; i++)
+            {
+                totalSum += args[i] * Math.pow(x, args.length - i - 1);
+            }
+            return totalSum;
+        };
+    }
+    else return null;
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: "else" is redundant due to "return"

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -104,7 +123,21 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+    return function() {
+        for(let i = 1; i <= attempts; i++) {
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js:125]: It is necessary to follow the generally accepted style of indentation from operators.

File: task/09-functions-n-closures-tasks.js [hunk 4]
```diff
@@ -104,7 +123,21 @@ function memoize(func) {
...
+            try 
+            {
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: Curly bracket should be opened on the same line as function or expression declaration

File: task/09-functions-n-closures-tasks.js [hunk 5]
```diff
@@ -171,7 +214,10 @@ function partialUsingArguments(fn) {
  *   getId10() => 11
  */
 function getIdGeneratorFunction(startFrom) {
-    throw new Error('Not implemented');
+    return () => 
+    {
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: Curly brackets are redundant in this arrow function.

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

File: task/09-functions-n-closures-tasks.js (hunk 4 above)
Comment (MikhailPyshnik) [reply task/09-functions-n-closures-tasks.js]: Fixed

File: task/09-functions-n-closures-tasks.js (hunk 5 above)
Comment (MikhailPyshnik) [reply task/09-functions-n-closures-tasks.js]: Fixed

File: task/09-functions-n-closures-tasks.js (hunk 3 above)
Comment (MikhailPyshnik) [reply task/09-functions-n-closures-tasks.js:125]: I don't understand your comment

File: task/09-functions-n-closures-tasks.js (hunk 2 above)
Comment (MikhailPyshnik) [reply task/09-functions-n-closures-tasks.js]: Fixed

File: task/09-functions-n-closures-tasks.js (hunk 1 above)
Comment (MikhailPyshnik) [reply task/09-functions-n-closures-tasks.js]: Fixed

File: task/09-functions-n-closures-tasks.js [hunk 6]
```diff
@@ -214,8 +209,7 @@ function partialUsingArguments(fn) {
  *   getId10() => 11
  */
 function getIdGeneratorFunction(startFrom) {
-    return () => 
-    {
+    return () => {
         return startFrom++;
     }
```
Comment (vramaniuk) [inline task/09-functions-n-closures-tasks.js]: @KostyaKov meant that it had to be `return () => startFrom++;`

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**